pkg/webhook/owner_reference/patching.go

```diff
@@ -81,17 +85,42 @@ func (r *handler) OnCreate(ctx context.Context, req admission.Request, clt clien
 
 	}
 
-	tl := &v1alpha1.TenantList{}
-	if err := clt.List(ctx, tl, client.MatchingFieldsSelector{
-		Selector: fields.OneTermEqualSelector(".spec.owner", req.UserInfo.Username),
-	}); err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
+	// assigning namespace to Tenant in case of --force-tenant-prefix flag enabled
+	if r.forceTenantPrefix {
```

Comment:
Do we really need this? We already have the `tenant_prefix` webhook.

Reply:
As i understand, `tenant_prefix` is just validating webhook. 
I've modified logic in this webhook to work like this:
- if force-tenant-prefix is enabled, extract tenantName from Namespace name and do simple lookup.
- if not - find first matching tenant either from User or from Group kind

Reply:
As I understand, if `--force-tenant-prefix` is true, the name of the tenant is already in the namespace, i.e. `tenant-namespace` so do a simple lookup in that case.

If `--force-tenant-prefix` is false, we have to do a more complex lookup to find which tenant the namespace belong to.